pkg/registries/factory_impl.go

```diff
@@ -26,7 +28,19 @@ func (r *registryWithDataSource) Source() *storage.ImageIntegration {
 }
 
 func (e *factoryImpl) CreateRegistry(source *storage.ImageIntegration) (types.ImageRegistry, error) {
-	creator, exists := e.creators[source.GetType()]
+	var creator Creator
+	var exists bool
+
+	if !source.Autogenerated || e.disableRepoListForAll {
+		creator, exists = e.creatorsWithoutRepoList[source.GetType()]
+	}
+
+	// If a creator was previously not found or not applicable, attempt to
```

Comment:
not going to block this PR over it, but if you are already going to make changes, then `not found nor applicable`